README.md

```diff
@@ -6,11 +6,20 @@ GitHub Action for creating a GitHub App installation access token.
 
 ## Usage
 
-In order to use this action, you need to:
-
-1. [Register new GitHub App](https://docs.github.com/apps/creating-github-apps/setting-up-a-github-app/creating-a-github-app)
-2. [Store the App's ID in your repository environment variables](https://docs.github.com/actions/learn-github-actions/variables#defining-configuration-variables-for-multiple-workflows) (example: `APP_ID`)
-3. [Store the App's private key in your repository secrets](https://docs.github.com/actions/security-guides/encrypted-secrets?tool=webui#creating-encrypted-secrets-for-a-repository) (example: `PRIVATE_KEY`)
+In order to use this action, you must first create a GitHub App. This is required in order to give create-github-app-token the necessary permissions to be able to generate tokens.
+
+[See this page](https://docs.github.com/apps/creating-github-apps/setting-up-a-github-app/creating-a-github-app) for more details on how to register a GitHub App. Here are ways that you can create the app:
+- [Use this link to create the GitHub app in your account](https://github.com/settings/apps/new?url=https://github.com/actions/create-github-app-token&webhook_active=false&public=false&contents=read&metadata=read)
+- [Use this link to create the GitHub app in your organization](https://github.com/organizations/:org/settings/apps/new?url=https://github.com/actions/create-github-app-token&webhook_active=false&public=false&contents=read&metadata=read) (note: link will be dead and you must replace `:org` with your organization name)
+- Here are the requirements if you manually create the app:
+    1. Uncheck `Webhook` > `Active` checkbox
+    2. Mark two permissions under `Repository permissions` as read-only: `Contents` and `Metadata`
```

Comment:
Permissions really depend on your use case. `Metadata` gets set automatically. I would remove this point, and remove the permissions from the URL query parameters above

Reply:
> Permissions really depend on your use case.

Thanks for the review! Two questions:
1. Even if people may need to add more permissions, does this URL represent the minimal level of permissions that they would need?
2. What other permissions could be needed and why? (I ask so I can see if these can be added to README.md).

This is something that GitHub ARC is really clear about and it would help to make create-github-app-token clear to save people team. If this means that I need to separate this new section into a "minimum necessary to get actions/checkout working", that's fine with me. Getting GitHub Enterprise submodules requires using this app, so that is my use case.

Reply:
> 1. Even if people may need to add more permissions, does this URL represent the minimal level of permissions that they would need?

Apps without any permissions still have access to everything that is public, even on accounts they are not installed on

> 2. What other permissions could be needed and why?

Pretty much every permission can be useful for something, I don't think we can make a complete list, there is not really a limit of what folks can do with apps, pretty much what users can do as well.

> GitHub ARC

What is GitHub ARC?

> If this means that I need to separate this new section into a "minimum necessary to get actions/checkout working", that's fine with me. Getting GitHub Enterprise submodules requires using this app, so that is my use case.

How about we add some common use cases? We can just start with the one(s) you have. In order to checkout a private/internal repository, the app needs the permission metadata:read, contents:read as you currently have it in your changes.

Reply:
> What is GitHub ARC?

It's this: https://github.com/actions/actions-runner-controller . They use an app too to get permissions